src/com/ikhokha/techcheck/CommentMetricProcessor.java

```diff
@@ -0,0 +1,14 @@
+package com.ikhokha.techcheck;
+
+public abstract class CommentMetricProcessor {
+    private String key;
+    public  CommentMetricProcessor(String key){
+    this.key=key;
+ 
+    }
+   abstract boolean results(String line);
+   String getKey(){
+        return this.key;
+    }
+    
+}
```

Comment:
This abstract class is used for each metric classes to extend it basic functionality